spark-connect/client/src/test/scala-spark-master/io/delta/connect/tables/RemoteSparkSession.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This file contains code from the Apache Spark project (original license above).
+ * It contains modifications, which are licensed as follows:
+ */
+
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.tables
+
+import org.scalatest.{BeforeAndAfterAll, Suite}
+
+import org.apache.spark.sql.SparkSession
+
+/**
+ * An util class to start a local Delta Connect server in a different process for local E2E tests.
+ * Pre-running the tests, the Delta Connect artifact needs to be built using e.g. `build/sbt
+ * assembly`. It is designed to start the server once but shared by all tests. It is equivalent to
+ * use the following command to start the connect server via command line:
+ *
+ * {{{
+ * bin/spark-shell \
+ * --jars `ls spark-connect/server/target/**/delta-connect-server-assembly*SNAPSHOT.jar | paste -sd ',' -` \
+ * --conf spark.plugins=org.apache.spark.sql.connect.SparkConnectPlugin
+ * --conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension
+ * --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog
+ * }}}
+ *
+ * Set system property `delta.test.home` or env variable `DELTA_HOME` if the test is not executed
+ * from the Delta project top folder.
+ */
+trait RemoteSparkSession extends BeforeAndAfterAll { self: Suite =>
+
+  private val serverPort = 15003
```

Comment:
@andreaschat-db I guess it's because in https://github.com/apache/spark/blob/master/connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/config/ConnectCommon.scala#L19, we have `private[sql] object ConnectCommon`, while we are using package `io.delta.tables`, to link with `spark-connect/client/src/main/scala-spark-master/io/delta/connect/tables/DeltaTable.scala`

Reply:
I see. It looks that we need to do some refactoring later on.